File: backend/api/src/main/java/com/locuspark/api/service/report/export/format/PaymentMethodLabel.java
package com.locuspark.api.service.report.export.format;

import com.locuspark.api.enums.PaymentMethod;

public final class PaymentMethodLabel {

    private PaymentMethodLabel() {
    }

    public static String label(PaymentMethod method) {
        if (method == null) {
            return "—";
        }
        return switch (method) {
            case DINHEIRO -> "Dinheiro";
            case PIX -> "Pix";
            case CARD_CREDIT -> "Cartão de Crédito";
            case CARD_DEBIT -> "Cartão de Débito";
        };
    }
}

File: frontend/src/app/shared/utils/payment-method-label.ts
import { PaymentMethod } from '../../core/domains/ticket/ticket.types';

export const PAYMENT_METHOD_LABELS: Record<PaymentMethod, string> = {
  DINHEIRO: 'Dinheiro',
  PIX: 'PIX',
  CARD_CREDIT: 'Cartão de Crédito',
  CARD_DEBIT: 'Cartão de Débito',
};

export function paymentMethodLabel(method: PaymentMethod): string {
  return PAYMENT_METHOD_LABELS[method];
}
